onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -485,34 +531,14 @@ TensorrtExecutionProvider::TensorrtExecutionProvider(const TensorrtExecutionProv
     runtime_ = tensorrt_ptr::unique_pointer<nvinfer1::IRuntime>(nvinfer1::createInferRuntime(GetTensorrtLogger()));
   }
 
-  const std::string engine_decryption_enable_env = onnxruntime::GetEnvironmentVar(tensorrt_env_vars::kDecryptionEnable);
-  if (!engine_decryption_enable_env.empty()) {
-    engine_decryption_enable_ = (std::stoi(engine_decryption_enable_env) == 0 ? false : true);
-  }
-
   if (engine_decryption_enable_) {
-    std::string engine_decryption_lib_path = onnxruntime::GetEnvironmentVar(tensorrt_env_vars::kDecryptionLibPath);
-    LIBTYPE handle = OPENLIB(engine_decryption_lib_path.c_str());
+    LIBTYPE handle = OPENLIB(engine_decryption_lib_path_.c_str());
     if (handle == nullptr) {
       ORT_MAKE_STATUS(ONNXRUNTIME, EP_FAIL,
-                      "TensorRT EP could not open shared library from " + engine_decryption_lib_path);
+                      "TensorRT EP could not open shared library from " + engine_decryption_lib_path_);
```

Comment:
constructor can't return status, so line 537 looks wrong?
wrap it in ORT_THROW_IF_ERROR ?